doc/en/historical-notes.rst

```diff
@@ -4,6 +4,114 @@ Historical Notes
 This page lists features or behavior from previous versions of pytest which have changed over the years. They are
 kept here as a historical note so users looking at old code can find documentation related to them.
 
+
+.. _marker-revamp:
+
+Marker revamp and iteration
+---------------------------
+
+.. versionchanged:: 3.6
+
+pytest's marker implementation traditionally worked by simply updating the ``__dict__`` attribute of functions to cumulatively add markers. As a result, markers would unintentionally be passed along class hierarchies in surprising ways. Further, the API for retrieving them was inconsistent, as markers from parameterization would be stored differently than markers applied using the ``@pytest.mark`` decorator and markers added via ``node.add_marker``.
+
+This state of things made it technically next to impossible to use data from markers correctly without having a deep understanding of the internals, leading to subtle and hard to understand bugs in more advanced usages.
+
+Depending on how a marker got declared/changed one would get either a ``MarkerInfo`` which might contain markers from sibling classes,
+``MarkDecorators`` when marks came from parameterization or from a ``node.add_marker`` call, discarding prior marks. Also ``MarkerInfo`` acts like a single mark, when it in fact represents a merged view on multiple marks with the same name.
+
+On top of that markers were not accessible the same way for modules, classes, and functions/methods.
+In fact, markers were only accessible in functions, even if they were declared on classes/modules.
+
+A new API to access markers has been introduced in pytest 3.6 in order to solve the problems with the initial design, providing :func:`_pytest.nodes.Node.iter_markers` method to iterate over markers in a consistent manner and reworking the internals, which solved great deal of problems with the initial design.
+
+
+.. _update marker code:
+
+Updating code
+~~~~~~~~~~~~~
+
+The old ``Node.get_marker(name)`` function is considered deprecated because it returns an internal ``MarkerInfo`` object
+which contains the merged name, ``*args`` and ``**kwargs`` of all the markers which apply to that node.
```

Comment:
But often this is just what you want - when checking if something has been marked, no?
(I can see that it makes sense to improve things, but remember having to change this in several places, although it would have been fine in general.)

Reply:
But we can't know, and given that `get_marker` and `get_closest_marker` returned different objects with different semantics, I think it was the right call by @RonnyPfannschmidt